### PR TITLE
Add discord handler

### PR DESCRIPTION
**Pull request by HunnySajid**
related Task : https://github.com/roots-id/roots-react-native/issues/21

ChangeLog:
- Removed sample and used faker to populate credential details
- Improved redux structure to dynamically create (dummy) credentials 

https://user-images.githubusercontent.com/19255438/203665189-17f42e6c-146c-435c-ad50-5f28c5777c8b.mov

### CODE REVIEW

**Comment by 2byrds** [comment]
Great stuff @HunnySajid!
This is close enough to some mocks i did yesterday that I think we can stick with it like you have it, but in case you want a little more context, here those are
See the slides here https://docs.google.com/presentation/d/15gq16k5ZRKe99rOhVuoNrEvksd7vViLkchTKlAkQSwM/edit?usp=sharing
 One thing that I'd like changed, when you accept or deny the credential, the message should show in the chat where you were offered the credential.  We only show messages in your own chat when the interaction wasn't with another person (you self-issued a demo credential).

File: src/store/thunks/wallet.ts [hunk 1]
```diff
@@ -1,24 +1,345 @@
...
+  async (credential, thunkAPI) => {
+    const { dispatch, getState } = thunkAPI;
+    const rootsHelper = getRootsHelperContact(getState());
+    const currentUser = getCurrentUserContact(getState());
+    const cred = (await dispatch(addCredentialToList(credential))).payload;
+    dispatch(
+      addMessage({
+        chatId: currentUser._id,
+        message: sendMessage(
+          currentUser._id,
```
Comment (2byrds) [inline src/store/thunks/wallet.ts]: this should post in the chat where you were offered the credential.

File: src/store/thunks/wallet.ts [hunk 2]
```diff
@@ -1,24 +1,345 @@
...
+    const { dispatch, getState } = thunkAPI;
+    const rootsHelper = getRootsHelperContact(getState());
+    const currentUser = getCurrentUserContact(getState());
+    dispatch(
+      addMessage({
+        chatId: currentUser._id,
+        message: sendMessage(
+          currentUser._id,
+          rootsHelper?._id,
+          `Discord social credential denied!`,
```
Comment (2byrds) [inline src/store/thunks/wallet.ts:336]: This should be posted in the chat where the credential was offered.

**Review by 2byrds — CHANGES_REQUESTED** [review]

**Comment by HunnySajid** [comment]
@2byrds  Updated the chatId for relevant message after credential acceptance/denial

Screenshot 2022-11-24 at 10 26 41 PM


Screenshot 2022-11-24 at 10 25 50 PM

**Comment by HunnySajid** [comment]
@2byrds  regarding the credentials flow diagram you shared, I will look into it more thoroughly over the weekend and will create MR if any changes are needed. 
I also have reviewed some open source projects regarding dids and chat based interfaces. I will check if we can also adopt any practises from there.

**Comment by 2byrds** [comment]
Looking forward to hearing your ideas. I'm just getting back from our holiday today. I'll be around to meet tomorrow if you can.  We have tons of nonSSI UI/UX work we need to push forward over the next 6 months.

**Comment by 2byrds** [comment]
> @2byrds  Updated the chatId for relevant message after credential acceptance/denial
> 
> Screenshot 2022-11-24 at 10 26 41 PM
> 
> 
> Screenshot 2022-11-24 at 10 25 50 PM
> 

Looks great @HunnySajid

**Comment by 2byrds** [comment]
confirmed on android
image
confirmed on iphone
image
confirmed on web
image

**Pull request merged**